src/vivarium_gates_mncnh/components/mortality.py

```diff
@@ -163,3 +166,120 @@ def get_proportional_case_fatality_rates(self, simulants: pd.DataFrame) -> pd.Da
             )
 
         return simulants
+
+
+class NeonatalMortality(Component):
+    """A component to handle neonatal mortality."""
+
+    ##############
+    # Properties #
+    ##############
+
+    @property
+    def configuration_defaults(self) -> dict[str, Any]:
+        return {
+            self.name: {
+                "data_sources": {
+                    "probability_death_by_age_group": self.load_probability_death_by_age_group,
+                    "life_expectancy": self.load_life_expectancy_data,
+                }
+            }
+        }
+
+    @property
+    def columns_created(self) -> list[str]:
+        return [COLUMNS.CHILD_CAUSE_OF_DEATH, COLUMNS.CHILD_YEARS_OF_LIFE_LOST]
+
+    @property
+    def columns_required(self) -> list[str]:
+        return [
+            COLUMNS.CHILD_AGE,
+            COLUMNS.CHILD_ALIVE,
+        ]
+
+    #####################
+    # Lifecycle methods #
+    #####################
+
+    def setup(self, builder: Builder) -> None:
+        self._sim_step_name = builder.time.simulation_event_name()
+        self.randomness = builder.randomness.get_stream(self.name)
+        self.location = get_location(builder)
+
+        self.probability_death_by_age_group = builder.value.register_value_producer(
+            "probability_death_by_age_group",
+            source=self.lookup_tables["probability_death_by_age_group"],
+            component=self,
+            requires_columns=get_lookup_columns(
+                [self.lookup_tables["probability_death_by_age_group"]]
+            ),
+        )
+
+    def on_initialize_simulants(self, pop_data: SimulantData) -> None:
+        pop_update = pd.DataFrame(
+            {
+                COLUMNS.CHILD_CAUSE_OF_DEATH: "not_dead",
+                COLUMNS.CHILD_YEARS_OF_LIFE_LOST: 0.0,
+            },
+            index=pop_data.index,
+        )
+        self.population_view.update(pop_update)
+
+    def on_time_step(self, event: Event) -> None:
+        if self._sim_step_name() not in [
+            SIMULATION_EVENT_NAMES.EARLY_NEONATAL_MORTALITY,
+            SIMULATION_EVENT_NAMES.LATE_NEONATAL_MORTALITY,
+        ]:
+            return
+
+        pop = self.population_view.get(event.index)
+        alive_children = pop.loc[pop[COLUMNS.CHILD_ALIVE] == "alive"]
+
+        # Update age of children to get correctlookup values
+        if self._sim_step_name() == SIMULATION_EVENT_NAMES.EARLY_NEONATAL_MORTALITY:
+            pop.loc[alive_children.index, COLUMNS.CHILD_AGE] = 7 / 365.0
+        else:
+            pop.loc[alive_children.index, COLUMNS.CHILD_AGE] = 28 / 365.0
+        self.population_view.update(pop)
+
+        # Determine which neonates die and update metadata
+        dead_idx = self.randomness.filter_for_probability(
+            alive_children.index,
+            self.probability_death_by_age_group(alive_children.index),
+            f"{self._sim_step_name}_neonatal_mortality_choice",
```

Comment:
We actually don't need to add the time-step name here (though it won't hurt anything to do so), since we will get different random numbers anyway, due to the fact that it is being called the second time on a different time-step.